### DISCUSSION TITLE
direnv support

### DISCUSSION THREAD

**Discussion by deifactor** [helix-editor/helix#7975] [Ideas] [answered]
I use the nix approach of "one direnv per project", which in turn means I need one helix instance per project. I'd like it if helix could pick up the envrc from the LSP root and use that when running the server/formatter/etc.

(There's also the question of what to do when go-to-definition takes you outside of that project. For example, with a Rust project ~/.cargo will generally not be in your direnv.)

Apologies if this is a dupe; I thought I filed this recently but I can't seem to find it on the tracker.

**Comment by pascalkuthe** [comment, marked as answer]
This is too specific, we generally don't integrate with third party tools directly. Direnv should workas long as you start helix in a working directory below the direnv file. This is what you would normally do anyway.

**Reply by deifactor** [reply]
Hm, okay. I'm more used to the emacs-ish "you have one editor you leave open forever and do everything in that" model, but it sounds like that's not the helix approach? Fair enough. (And yeah, it works if I start helix in the directory.)
